net/Socket.hpp

```diff
@@ -501,6 +483,30 @@ class Socket
         }
     }
 
+    /// Close the socket FD.
+    /// Internal implementation, always private.
+    void closeFD()
+    {
+        // explicit closeFD called, or initial createSocket failure
+        if (_fd < 0)
+            return;
```

Comment:
Could have some tracing, should this occur, there is a bug in calling code.

Reply:
This will get called more than once when we close explicitly and when we destroy (it's called from the destructor in this patch, line 164, above).